packages/plugin-core/src/WSUtilsV2.ts

```diff
@@ -268,4 +271,78 @@ export class WSUtilsV2 implements IWSUtilsV2 {
     const fnameWithExtension = `${fname}.md`;
     return this.openFileInEditorUsingFullFname(vault, fnameWithExtension);
   }
+
+  async findDuplicateNoteFromDocument(document: vscode.TextDocument) {
+    const fsPath = document.uri.fsPath;
+    const { vaults, wsRoot, engine } = this.extension.getDWorkspace();
+    let vault;
+    try {
+      vault = VaultUtils.getVaultByFilePath({
+        vaults,
+        wsRoot,
+        fsPath,
+      });
+    } catch (error: any) {
+      // document doesn't exist in workspace.
+      return;
+    }
+
+    // we do this because the note in document would _not_ be in our store
+    // if it is a duplicate note.
+    const currentNote = file2Note(fsPath, vault);
+
+    // find the potentially-duplicate note that's currently in our store.
+    const noteById = await engine.getNote(currentNote.id);
+
+    let hasDuplicate = false;
+
+    if (noteById !== undefined) {
+      if (currentNote.id === noteById.id) {
+        // id of note in store and from document is the same. we _might_ have hit a duplicate.
+        if (VaultUtils.isEqualV2(currentNote.vault, noteById.vault)) {
```

Comment:
nitpick: Could simplify this as

```ts
const hasDuplicate = !VaultUtils.isEqualV2(currentNote.vault, noteById.vault) || currentNote.fname !== noteById.fname;
```

Reply:
I did start with one statement but unfurled it here for readability. I think it's clearer this way.